### PR TITLE
FAI-389: Score Cards: Integrate with Online channel

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.redhat.com/browse/FAI-389

Scorecard Editor added to "online" channel.

Screenshot from 2021-03-22 20-12-51

Screenshot from 2021-03-22 20-12-58

@tiagobento Who/where did the SVG for the logos as I've just copied the DMN one (see logo on second screenshot above).

I removed "Share -> Download SVG" for the PMML editor as it does not support it.

This PR also includes a simple integration test for the PMML editor.

### CODE REVIEW

File: packages/online-editor/it-tests/integration/TrySampleTest.ts [hunk 1]
```diff
@@ -15,118 +15,138 @@
  */
 
 describe("Try sample test", () => {
-    beforeEach(() => {
```
Comment (manstis) [inline packages/online-editor/it-tests/integration/TrySampleTest.ts:18]: Mainly formatting changes (it looks like 4 spaces -> 2 spaces)

File: packages/online-editor/it-tests/integration/TrySampleTest.ts [hunk 2]
```diff
@@ -15,118 +15,138 @@
...
+        expect($dataTypes.eq(11)).text("Back_End_Ratio");
+        expect($dataTypes.eq(12)).text("Front_End_Ratio");
+        expect($dataTypes.eq(13)).text("Qualification");
+        expect($dataTypes.eq(14)).text("Credit_Score");
+        expect($dataTypes.eq(15)).text("Loan_Qualification");
+      });
+    });
+  });
+
+  it("should create PMML sample", () => {
```
Comment (manstis) [inline packages/online-editor/it-tests/integration/TrySampleTest.ts:128]: New integration test!

File: packages/online-editor/src/editor/EditorToolbar.tsx [hunk 3]
```diff
@@ -128,14 +128,19 @@ export function EditorToolbar(props: Props) {
       >
         {i18n.editorToolbar.copySource}
       </DropdownItem>,
-      <DropdownItem key={`dropdown-${dropdownId}-download-svg`} component="button" onClick={props.onPreview}>
-        {i18n.editorToolbar.downloadSVG}
-      </DropdownItem>,
+      <React.Fragment key={`dropdown-${dropdownId}-fragment-download-svg`}>
+        {(fileExtension.toLowerCase() === i18n.names.bpmn.toLowerCase() ||
```
Comment (manstis) [inline packages/online-editor/src/editor/EditorToolbar.tsx]: Only show "Download SVG" for DMN and BPMN.

File: packages/online-editor/static/images/pmml_kogito_logo.svg [hunk 4]
```diff
@@ -0,0 +1,99 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (manstis) [inline packages/online-editor/static/images/pmml_kogito_logo.svg]: This is a copy and paste of the DMN logo... where can I get a PMML version made!?

File: packages/online-editor/src/editor/EditorToolbar.tsx [hunk 5]
```diff
@@ -128,14 +128,19 @@ export function EditorToolbar(props: Props) {
...
+          fileExtension.toLowerCase() === i18n.names.dmn.toLowerCase()) && (
```
Comment (tiagobento) [inline packages/online-editor/src/editor/EditorToolbar.tsx]: This won't work for `bpmn2` files. Also, not a good idea to compare i18n strings with a file extension. I would create an array called "svgUnsupportedFileExtensions" and make it have the "pmml" extension on this array. Then I would compare the current file extension with the ones on this array and check if the button should be displayed or not.

File: packages/online-editor/static/images/pmml_kogito_logo.svg (hunk 4 above)
Comment (tiagobento) [reply packages/online-editor/static/images/pmml_kogito_logo.svg]: I honestly don't know 😩

cc @ederign @porcelli

**Review by tiagobento — CHANGES_REQUESTED** [review]
Single comment inline!

File: packages/online-editor/static/images/pmml_kogito_logo.svg (hunk 4 above)
Comment (ederign) [reply packages/online-editor/static/images/pmml_kogito_logo.svg]: Originally, Kyle created the logos for us. Maybe talking with Liz?

File: packages/online-editor/it-tests/integration/TrySampleTest.ts [hunk 6]
```diff
@@ -15,118 +15,138 @@
...
+    // clicks DMN Try Sample
```
Comment (tomasdavidorg) [inline packages/online-editor/it-tests/integration/TrySampleTest.ts]:
```suggestion
    // clicks PMML Try Sample
```

File: packages/online-editor/it-tests/integration/TrySampleTest.ts [hunk 7]
```diff
@@ -15,118 +15,138 @@
...
+    cy.get("[data-ouia-component-id='try-pmml-sample-button']").click();
+
+    // checks editor title name
+    cy.get("[aria-label='Edit file name']").should("have.value", "sample");
+
+    // waits until loading dialog disappears
+    cy.loadEditor();
+
+    cy.getEditor().within(() => {
```
Comment (tomasdavidorg) [inline packages/online-editor/it-tests/integration/TrySampleTest.ts:138]:
```suggestion
    cy.getEditor().within(() => {
      cy.get(".characteristics-container div > strong").then($dataTypes => {
        expect($dataTypes).length(3);
        expect($dataTypes.eq(0)).text("departmentScore");
        expect($dataTypes.eq(1)).text("ageScore");
        expect($dataTypes.eq(2)).text("incomeScore");
      });
```

File: packages/online-editor/it-tests/integration/TrySampleTest.ts [hunk 8]
```diff
@@ -15,118 +15,138 @@
...
+      // Open and close PMML DataDictionary modal
+      cy.get("[data-title='DataDictionary']").click();
+      cy.get("[data-title='DataDictionaryModalClose']").click();
```
Comment (tomasdavidorg) [inline packages/online-editor/it-tests/integration/TrySampleTest.ts]:
```suggestion
      cy.get("[data-title='DataDictionary']").click();
      cy.get(".data-type-item__name").then($dataTypes => {
        expect($dataTypes).length(4);
        expect($dataTypes.eq(0)).text("department");
        expect($dataTypes.eq(1)).text("age");
        expect($dataTypes.eq(2)).text("income");
        expect($dataTypes.eq(3)).text("overallScore");
      });
      cy.get("[data-title='DataDictionaryModalClose']").click();
```

File: packages/online-editor/it-tests/integration/TrySampleTest.ts [hunk 9]
```diff
@@ -15,118 +15,138 @@
...
+
+      // Open and close PMML MiningSchema modal
+      cy.get("[data-title='MiningSchema']").click();
+      cy.get("[data-title='MiningSchemaModalClose']").click();
```
Comment (tomasdavidorg) [inline packages/online-editor/it-tests/integration/TrySampleTest.ts]:
```suggestion
      cy.get("[data-title='MiningSchema']").click();
      cy.get(".mining-schema-list__item__name").then($miningSchema => {
        expect($miningSchema).length(4);
        expect($miningSchema.eq(0)).text("department");
        expect($miningSchema.eq(1)).text("age");
        expect($miningSchema.eq(2)).text("income");
        expect($miningSchema.eq(3)).text("overallScore");
      });
      cy.get("[data-title='MiningSchemaModalClose']").click();
```

File: packages/online-editor/it-tests/integration/TrySampleTest.ts [hunk 10]
```diff
@@ -15,118 +15,138 @@
...
-                expect($dataTypes.eq(11)).text("Back_End_Ratio");
-                expect($dataTypes.eq(12)).text("Front_End_Ratio");
-                expect($dataTypes.eq(13)).text("Qualification");
-                expect($dataTypes.eq(14)).text("Credit_Score");
-                expect($dataTypes.eq(15)).text("Loan_Qualification");
-            });
-        });
+      // Open and close PMML Outputs modal
+      cy.get("[data-title='Outputs']").click();
+      cy.get("[data-title='OutputsModalClose']").click();
```
Comment (tomasdavidorg) [inline packages/online-editor/it-tests/integration/TrySampleTest.ts]:
```suggestion
      cy.get("[data-title='Outputs']").click();
      cy.get(".outputs-container div > strong").then($outputs => {
        expect($outputs).length(4);
        expect($outputs.eq(0)).text("Final Score");
        expect($outputs.eq(1)).text("Reason Code 1");
        expect($outputs.eq(2)).text("Reason Code 2");
        expect($outputs.eq(3)).text("Reason Code 3");
      });
      cy.get("[data-title='OutputsModalClose']").click();
```

**Review by tomasdavidorg — CHANGES_REQUESTED** [review]
@manstis Thank you so much for the test 🙂. Would you please include the basic text checks I suggested? Thanks.

File: packages/online-editor/it-tests/integration/TrySampleTest.ts (hunk 6 above)
Comment (manstis) [reply packages/online-editor/it-tests/integration/TrySampleTest.ts]: Fixed in new commit.

File: packages/online-editor/it-tests/integration/TrySampleTest.ts (hunk 7 above)
Comment (manstis) [reply packages/online-editor/it-tests/integration/TrySampleTest.ts:138]: Fixed in new commit.

File: packages/online-editor/it-tests/integration/TrySampleTest.ts (hunk 8 above)
Comment (manstis) [reply packages/online-editor/it-tests/integration/TrySampleTest.ts]: Fixed in new commit.

File: packages/online-editor/it-tests/integration/TrySampleTest.ts (hunk 9 above)
Comment (manstis) [reply packages/online-editor/it-tests/integration/TrySampleTest.ts]: Fixed in new commit.

File: packages/online-editor/it-tests/integration/TrySampleTest.ts (hunk 10 above)
Comment (manstis) [reply packages/online-editor/it-tests/integration/TrySampleTest.ts]: Fixed in new commit.

File: packages/online-editor/src/editor/EditorToolbar.tsx (hunk 5 above)
Comment (manstis) [reply packages/online-editor/src/editor/EditorToolbar.tsx]: Yep, OK. Fair point. I've re-implemented.

File: packages/online-editor/static/images/pmml_kogito_logo.svg (hunk 4 above)
Comment (manstis) [reply packages/online-editor/static/images/pmml_kogito_logo.svg]: I managed to cobble something together:

Screenshot from 2021-03-23 10-57-27

**Review by kelvah — APPROVED** [review]

**Comment by manstis** [comment]
Hi @tiagobento @ederign @tomasdavidorg this PR is "good to go"... it just needs your final approvals.

**Comment by porcelli** [comment]
+1 from me. we may need a better UX in the near future (especially once this is more 'official' and have at least a domain URL)

**Review by porcelli — APPROVED** [review]

**Review by tomasdavidorg — APPROVED** [review]
Looks good to me. Thanks 🙂 .

**Comment by tiagobento** [comment]
@manstis Hi! Just noticed a minor issue with this PR. The PMML Editor doesn't have "Embed" capabilities yet, so we should disable this option on the Kebab menu at the top-right corner...

**Comment by manstis** [comment]
@tiagobento Fixed (in new commit). Thank-you.

**Review by tiagobento — APPROVED** [review]

**Pull request merged**